cosmo/netboxclient.py

```diff
@@ -242,6 +251,43 @@ def get_data(self, device_config):
 class NetboxV4Strategy(NetboxStrategy):
 
     def get_data(self, device_config):
+        device_list = device_config['router'] + device_config['switch']
+
+        with Pool() as pool:
+            device_data = pool.apply_async(self.get_base_data, (device_list,))
```

Comment:
maybe, instead of managing the parallelization by hand:
- make a ParallelizableQuery abstract class with query and get properties
- make something that can manage threading outside of NetboxV4Strategy (can be contained in ParallelizableQuery, maybe? then would be ParallelQuery)
- make it chainable with either ParallelizableQuery.mergeWith(obj: ParallelizableQuery) or mergeWith(obj: json), so that it can be stitched in the order we want, by hand.